crates/api_common/src/request.rs

```diff
@@ -158,23 +152,21 @@ fn extract_opengraph_data(html_bytes: &[u8], url: &Url) -> Result<OpenGraphData,
 pub async fn extract_thumbnail_from_opengraph_data(
   url: &Url,
   opengraph_data: &OpenGraphData,
-  content_type: &Option<Mime>,
   generate_thumbnail: bool,
   context: &LemmyContext,
 ) -> Option<DbUrl> {
-  let is_image = content_type.as_ref().unwrap_or(&mime::TEXT_PLAIN).type_() == mime::IMAGE;
-  if generate_thumbnail && is_image {
```

Comment:
This was actually checking the thumbnail of the post.url, ie a youtube.com url. So this would give `is_image = false` even for a valid opengraph thumbnail.